### ISSUE TITLE
关于A0的设置以及\delta - Softmax 的公式设置

### ISSUE THREAD

**Issue by Duanclub1996** [chunhuiz/MPGE-RootRank-for-root-cause-diagnosis#1]
# 问题一 A0的设置
在文章中并未提及A0的设置及其原因，从代码中看A0 是一个全联通图，自身连接权重为2倍的与其他节点连接权重。\
如此设置A0的原因是什么？
```python
A0=torch.ones(args.n_e,args.n_e)+torch.eye(args.n_e)
A0=A0/torch.sum(A0,1)
```
且A0似乎只在初始化阶段进行了归一化，与文章中的公式似乎不符。
$$F_{out}=\sigma (D^{-1}AF_{in}W)$$

# 问题二 $\delta - Softmax$ 

文章中：\
$$\delta-\mathrm{Softmax}=a_{j}^{s} = \frac{e^{\delta a_j}}{\sum\limits_{i=1}^{J} e^{\delta a_j}}$$
$$\widetilde{\mathbf{A}}^{(l+1)}=\delta-\mathrm{Softmax}\left(\widetilde{\mathbf{A}}^{(l)}\right),l=1,\ldots,L$$

代码中：

$$\widetilde{\mathbf{A}}^{(l+1)}=\delta-\mathrm{Softmax}\left(\delta-\mathrm{Softmax}\left(\widetilde{\mathbf{A}}^{(l)}\right)\right),l=1,\ldots,L$$

是否以代码为准？

**Comment by Duanclub1996** [comment]
代码中训练损失函数：
```python
loss_reg=lamda*torch.sum(torch.abs(model.A0)) + (0.001)*torch.sum((model.A0)**2)
```

文章中：
$$\beta_1\sum_{i=1}^J\sum_{j=1}^J|a_{ij}|+\beta_2\parallel A \parallel_F^2 $$

Frobenius 范数:\
$$\parallel A \parallel_F^2 = \sqrt{\sum_{i=1}^J\sum_{j=1}^J|a_{ij}|^2}$$

是否在代码中缺少了开方？
